salesforce-connector/src/com/axonivy/connector/salesforce/auth/OAuth2Feature.java

```diff
@@ -0,0 +1,158 @@
+		oauth2.tokenSuffix(() -> GrantType.of(config).type);
+		context.register(oauth2, Priorities.AUTHORIZATION);
+		return true;
+	}
+
+	private static Response requestToken(AuthContext ctxt, OAuth2UriProperty uriFactory) {
+		FeatureConfig config = ctxt.config;
+		var authCode = ctxt.authCode();
+		var refreshToken = ctxt.refreshToken();
+		GrantType grant = GrantType.of(config);
+		if (authCode.isEmpty() && refreshToken.isEmpty() && GrantType.AUTH_CODE == grant) {
+			authError(config, uriFactory).withMessage("missing permission from user to act in his name.").throwError();
+		}
+
+		Form form = createTokenPayload(config, authCode, refreshToken);
+		var response = ctxt.target.request().accept(MediaType.WILDCARD).post(Entity.form(form));
+		return response;
+	}
+
+	static Form createTokenPayload(FeatureConfig config, Optional<String> authCode, Optional<String> refreshToken) {
+		Form form = new Form();
+		form.param("client_id", config.readMandatory(Property.APP_ID));
+		form.param("client_secret", config.readMandatory(Property.CLIENT_SECRET));
+		GrantType grant = GrantType.of(config);
+		form.param("grant_type", grant.type);
+		configureGrant(config, authCode, form, grant);
+		if (refreshToken.isPresent()) {
+			form.param("redirect_uri", ivyCallbackUri().toASCIIString());
+			form.param("refresh_token", refreshToken.get());
+			form.asMap().putSingle("grant_type", "refresh_token");
+		}
+		return form;
+	}
+
+	private static void configureGrant(FeatureConfig config, Optional<String> authCode, Form form, GrantType grant) {
+		switch (grant) {
+		case APPLICATION:
+			//form.param("scope", config.read(Property.SCOPE).orElse(Default.APP_SCOPE));
+			break;
+		case PASSWORD:
+			//form.param("scope", getPersonalScope(config));
+			form.param("username", config.readMandatory(Property.USER));
+			form.param("password", config.readMandatory(Property.PASS));
+			break;
+		default:
+		case AUTH_CODE:
+			//form.param("scope", getPersonalScope(config));
+			form.param("redirect_uri", ivyCallbackUri().toASCIIString());
+			authCode.ifPresent(code -> form.param("code", code));
+		}
+	}
+
+	private static BpmPublicErrorBuilder authError(FeatureConfig config, OAuth2UriProperty uriFactory) {
+		var uri = createMsAuthCodeUri(config, uriFactory);
+		return OAuth2RedirectErrorBuilder.create(uri).withMessage("Missing permission from user to act in his name.");
+	}
+
+	private static URI createMsAuthCodeUri(FeatureConfig config, OAuth2UriProperty uriFactory) {
+		return UriBuilder.fromUri(uriFactory.getUri("authorize"))
+				.queryParam("client_id", config.readMandatory(Property.APP_ID))
+				//.queryParam("scope", getPersonalScope(config)).queryParam("redirect_uri", ivyCallbackUri())
+				.queryParam("response_type", "code").queryParam("response_mode", "query").build();
+	}
+
+	private static URI ivyCallbackUri() {
+		return OAuth2CallbackUriBuilder.create().toUrl();
+	}
+
+//	private static String getPersonalScope(FeatureConfig config) {
```

Comment:
do we need this line code for investigating any issue? or can we remove it? @HoDacVu90